src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -255,23 +255,12 @@ static void registerBukkitEvents() {
 			if (e.equals(PlayerInteractAtEntityEvent.class) || e.equals(PlayerArmorStandManipulateEvent.class)) {
 				continue; // Ignore, registered above
 			}
-			
-			if (!containsSuperclass((Set) registeredEvents, e)) { // I just love Java's generics
+			if (!registeredEvents.contains(e)) { // Check if event is registered
```

Comment:
This will not work for something like `PlayerCommandPreprocessEvent` and `PlayerSignCommandPreprocessEvent`, because a listener for those will be registered twice, because only `PlayerCommandPreprocessEvent` implements the event HandlerList logic.

A fix for this requires a more elaborate solution, described in https://github.com/SkriptLang/Skript/issues/4705#issuecomment-1085851133

Reply:
I dont see the issue, when testing it seems it registeres the events correctly.

I tried testing the following things:

Test 1
---
When starting the server with the following skript code the events in the picture under gets registred:
```
on command:
   broadcast "Command"
    
on sign command:
    broadcast "Sign Command"
```
![billede](https://user-images.githubusercontent.com/75088349/179024209-25024bbc-e839-4efd-8bbd-0839e2f083e8.png)

When running command `/time set day` only the `on command` event gets trigered.
![billede](https://user-images.githubusercontent.com/75088349/179024942-da0fd555-e66a-45f2-9b57-7b1a1ce6f177.png)

And when clicking on the command sign the `on sign command` and `on command` event gets trigered.
![billede](https://user-images.githubusercontent.com/75088349/179025454-2c1f8586-3b25-4cfd-b70a-8f7c00ea978c.png)

Test 2
---
When starting the server with the following skript code the events in the picture under gets registred:
```
on command:
   broadcast "Command"
```
![billede](https://user-images.githubusercontent.com/75088349/179026380-6700d5b2-592d-4c5b-bdb0-0c67830eefaa.png)

When running command `/time set day` only the `on command` event gets trigered.
![billede](https://user-images.githubusercontent.com/75088349/179026691-ff1e031c-b01b-4746-a94e-e629fe52f686.png)

And when clicking on the command sign only the `on command` event gets trigered.
![billede](https://user-images.githubusercontent.com/75088349/179026964-f54db36c-0005-4acf-a1c8-78397585e0ed.png)

Test 3
---
When starting the server with the following skript code the events in the picture under gets registred:
```
on sign command:
    broadcast "Sign Command"
```
![billede](https://user-images.githubusercontent.com/75088349/179027565-6b00914f-fc05-4b99-b8b8-ccee5edc2440.png)

When running command `/time set day` nothing happends.

And when clicking on the command sign the `on sign command event gets trigered.
![billede](https://user-images.githubusercontent.com/75088349/179028500-18a57e7f-16c6-4425-81d6-a17fd13660a4.png)


**I also tested with [PlayerTradeEvent](https://jd.papermc.io/paper/1.17/io/papermc/paper/event/player/PlayerTradeEvent.html) and [PlayerPurchaseEvent](https://jd.papermc.io/paper/1.17/io/papermc/paper/event/player/PlayerPurchaseEvent.html)**

Code and commands used for testing
---
Here is the edited part of the SkriptEventHandler (added debug messages):
```java
if (!registeredEvents.contains(e)) { // Check if event is registered
	Skript.info("Register event: " + e);
	Bukkit.getPluginManager().registerEvent(e, listener, priority, executor, Skript.getInstance());
	registeredEvents.add(e);
}
Skript.info("registredEvents list: " + registeredEvents);
```
And here is my implementation of the **on sign command** event:
```java
Skript.registerEvent("Sign Command Event", SimpleEvent.class, PlayerSignCommandPreprocessEvent.class, new String[]{"sign command"});

EventValues.registerEventValue(PlayerSignCommandPreprocessEvent.class, Player.class, new Getter<Player, PlayerSignCommandPreprocessEvent>() {
	@Override
	public Player get(PlayerSignCommandPreprocessEvent e) {
		return e.getPlayer();		
        }
}, 0);
```
And here is the command i used to get a command sign
`/give @p minecraft:oak_sign{BlockEntityTag:{Text1:'{"text":"Command","clickEvent":{"action":"run_command","value":"time set day"}}'},display:{Name:'{"text":"Command"}'}}`

Reply:
Try this code:
```
on test command: # PlayerCommandPreprocessEvent
	broadcast "Command"
	make player shoot arrow # or any code that triggers a registered event

on test sign command: # PlayerSignCommandPreprocessEvent
	broadcast "Sign Command"

on shoot:
	broadcast "shoot"
```
```java
Skript.registerEvent("Sign Command Event", SimpleEvent.class, PlayerSignCommandPreprocessEvent.class, new String[]{"test sign command"});
Skript.registerEvent("Pre Command Event", SimpleEvent.class, PlayerCommandPreprocessEvent.class, new String[]{"test command"});
```
and click a sign / run a command:
![image](https://user-images.githubusercontent.com/29547183/179052409-ff121209-7a1b-4519-8e97-434ff167e01e.png)

The reason your previous tests worked, was because of line SkriptEventHandler line 69. 
This line is a hacky solution to the problem, to stop double event triggers. However, if you change the lastEvent (for example by making your script call an event the executor is registered to listen to) during the first execution, when it's called the second time, it won't prevent it from executing the trigger because lastEvent has changed.

Your solution doesn't work because both Events use the same HandlerList, so registering it for both will register the executor twice to that HandlerList, causing double firing.

Reply:
Checking if Skript has a listener registred for a event (as sugested in https://github.com/SkriptLang/Skript/issues/4705#issuecomment-1085851133) didn't work since Skript already have listeners registred for some events which causes SkriptEventHandler to not register a listener.

I tried making a workaround, which seems to work after testing.